test/js/util.test.ts

```diff
@@ -98,3 +98,22 @@ describe('numToString', () => {
         expect(numToString(0x4142434445)).toEqual('EDCB');
     });
 });
+
+describe('stringToNum', () => {
+    it('packs a zero byte into a string of all zeros', () => {
+        expect(stringToNum('\0\0\0\0')).toEqual(0x00);
+    });
+    it('packs a byte in the printable ASCII range into a zero-padded string',
+        () => {
+            expect(stringToNum('A\0\0\0')).toEqual(0x41);
+        });
+    it('packs a word into a string', () => {
+        expect(stringToNum('BA\0\0')).toEqual(0x4142);
+    });
+    it('packs a 32-bit value into a string', () => {
+        expect(stringToNum('DCBA')).toEqual(0x41424344);
+    });
+    it('ignores more than 4 character', () => {
+        expect(stringToNum('EDCBA')).toEqual(0x42434445);
+    });
+});
```

Comment:
I've been wondering about `numToString` and `stringToNum`. To me, they seem to obfuscate what's actually going on. Here's an example:
```TypeScript
    const prefix = new DataView(rawData);
    const signature = numToString(prefix.getInt32(OFFSETS.SIGNATURE, true));
    if (signature !== '2IMG') {
        throw new Error(`Unrecognized 2mg signature: ${signature}`);
    }
```
Let's assume the following bytes in `rawData`:
```none
32  49  4D 47 ...
```
Here, `prefix.getInt32(OFFSETS.SIGNATURE, true)` will return `0x474d4932` and then `numToString` pulls off characters from the low-order bits, shifting down each time.

Wouldn't it be more straightforward to do something like:
```TypeScript
const signature = new TextDecoder('ascii').decode(new Uint8Array(rawData, OFFSETS.SIGNATURE, 4));
```

Similarly for `stringToNum`, you should be able to produce the same effect with something like:
```TypeScript
const signature = new TextEncoder().encodeInto(
    myString, new Uint8Array(rawData, OFFSETS.SIGNATURE, 4).fill(0));
```

Granted, I haven't tested these, but it seems more straightforward.

Reply:
I restructured it a bit and I think it's pretty readable now.

Reply:
So new that Node 12 doesn't cut it. Which I guess is fine since it's been EOLed.